apps/files_versions/lib/Listener/FileEventsListener.php

```diff
@@ -402,6 +402,24 @@ private function getPathForNode(Node $node): ?string {
 			}
 		}
 
+		try {
+			$this->logger->debug('Failed to compute path for node', [
+				'node' => [
+					'path' => $node->getPath(),
+					'owner' => $owner,
+					'fileid' => $node->getId(),
+					'size' => $node->getSize(),
+					'mtime' => $node->getMTime(),
+				]
+			]);
+		} catch (NotFoundException) {
+			$this->logger->debug('Failed to compute path for node', [
+				'node' => [
+					'path' => $node->getPath(),
+					'owner' => $owner,
+				]
+			]);
+		}
```

Comment:
I don't understand which method call is supposed to throw that exception. Shouldn't the try catch wrap some other operation?

Reply:
If `$node` is a `NonExistingFile`, it only supports `getPath()`, all other methods throws `NotFoundException`

Reply:
Just to make it obvious then:

```suggestion
		if (!($node instanceof NonExistingFile)) {
			$this->logger->debug('Failed to compute path for node', [
				'node' => [
					'path' => $node->getPath(),
					'owner' => $owner,
					'fileid' => $node->getId(),
					'size' => $node->getSize(),
					'mtime' => $node->getMTime(),
				]
			]);
		} else {
			$this->logger->debug('Failed to compute path for node', [
				'node' => [
					'path' => $node->getPath(),
					'owner' => $owner,
				]
			]);
		}
```

Reply:
Could also be NonExistingFolder, I think

Reply:
@artonge Pushed that, please review.